win32/devdoc/job_object_helper.md

```diff
@@ -0,0 +1,46 @@
+# job_object_help requirements
+
+`job_object_help` is a module that wraps the Windows JobObject API, primarily for the purpose of limiting the amount of memory and CPU the current process can consume.
+
+## Exposed API
+```c
+    MOCKABLE_FUNCTION(, int, job_object_helper_limit_resources, uint32_t, percent_physical_memory, uint32_t, percent_cpu);
+```
+
+## job_object_helper_limit_resources
+```c
+    MOCKABLE_FUNCTION(, int, job_object_helper_limit_resources, uint32_t, percent_physical_memory, uint32_t, percent_cpu);
+```
+
+`job_object_helper_limit_resources` is an API that uses the job object API to limit the amount of physical memory and CPU that the current process can consume.
+
+**SRS_JOB_OBJECT_HELPER_18_001: [** `percent_physical_memory` is allowed to be `0`, which means "do not limit physical memory usage". **]**
+
+**SRS_JOB_OBJECT_HELPER_18_002: [** `percent_cpu` is allowed to be `0`, which means "do not limit CPU usage". **]**
+
+**SRS_JOB_OBJECT_HELPER_18_003: [** If `percent_physical_memory` and `percent_cpu` are both `0`, `job_object_helper_limit_resources` shall succeed and return `0`. **]**
+
+**SRS_JOB_OBJECT_HELPER_18_004: [** If `percent_physical_memory` is greater than `100`, `job_object_helper_limit_resources` shall fail and return a non-zero value. **]**
+
+**SRS_JOB_OBJECT_HELPER_18_005: [** If `percent_cpu` is greater than `100`, `job_object_helper_limit_resources` shall fail and return a non-zero value. **]**
+
+**SRS_JOB_OBJECT_HELPER_18_006: [** `job_object_helper_limit_resources` shall call `CreateJobObject` to create a new job object passing `NULL` for both `lpJobAttributes` and `lpName`. **]**
+
+**SRS_JOB_OBJECT_HELPER_18_007: [** `job_object_helper_limit_resources` shall call `GetCurrentProcess` to get the current process handle. **]**
+
+**SRS_JOB_OBJECT_HELPER_18_008: [** `job_object_helper_limit_resources` shall call `AssignProcessToJobObject` to assign the current process to the new job object. **]**
+
+**SRS_JOB_OBJECT_HELPER_18_009: [** If `percent_physical_memory` is not `0`, `job_object_helper_limit_resources` shall call `GlobalMemoryStatusEx` to get the total amount of physical memory in kb. **]**

```

Comment:
> If `percent_physical_memory` is not `0` [](http://example.com/codeflow?start=35&length=40)

003 says that the function will return. 

This check seems redundant

Reply:
same for 010

Reply:
and 011

Reply:
003 comes into play only if they are both `0`.

Reply:
rewritten to remove ambiguity.